### PR TITLE
Code Optimization - PART 7

### PR DESCRIPTION
**Pull request by oualib**

### CODE REVIEW

File: verticapy/vdataframe.py [hunk 1]
```diff
@@ -2541,7 +2524,7 @@ def aggregate(
                     for elem in agg
                 ]
                 query = (
```
Comment (mail4umar) [inline verticapy/vdataframe.py:2526]: there is query defined just before this.

**Review by mail4umar — APPROVED** [review]

File: verticapy/plot.py [hunk 2]
```diff
@@ -1995,39 +1961,49 @@ def compute_plot_variables(
...
+                (SELECT 
+                    {{}}
+                    TIMESTAMPADD('second',
+                                 {math.floor(h * idx)},
+                                 '{min_date}'::timestamp))"""
             if idx == 0:
-                query += "((SELECT /*+LABEL('plot.compute_plot_variables')*/ TIMESTAMPADD('second' , {}, '{}'::timestamp))".format(
-                    math.floor(h * idx), min_date
-                )
+                query += query_tmp.format("/*+LABEL('plot.compute_plot_variables')*/")
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/plot.py:2004]: there is a condition it is added if the condition is validated

File: verticapy/vdataframe.py (hunk 1 above)
Comment (oualib) [reply verticapy/vdataframe.py:2526]: yes but this one is using it again.

File: verticapy/plot.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/plot.py:2004]: there is a condition to do it

File: verticapy/plot.py [hunk 3]
```diff
@@ -1431,31 +1407,22 @@ def bubble(
...
-                ],
-                [
-                    min([item[2] for item in query_result]),
-                    max([item[2] for item in query_result]),
-                ],
+                        markersize=fun(size) / 100 + 15,
+                    )
+                ]
+                args_legends[1] += [fun([x[2] for x in query_result])]
+            leg1 = ax.legend(
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/plot.py:1424]: looks good to me

**Pull request merged**